jib-core/src/main/java/com/google/cloud/tools/jib/ncache/DefaultCacheStorageWriter.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.ncache;
+
+import com.google.cloud.tools.jib.blob.BlobDescriptor;
+import com.google.cloud.tools.jib.blob.Blobs;
+import com.google.cloud.tools.jib.hash.CountingDigestOutputStream;
+import com.google.cloud.tools.jib.image.DescriptorDigest;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.zip.GZIPOutputStream;
+
+/** Writes to the default cache storage engine. */
+class DefaultCacheStorageWriter {
+
+  private final DefaultCacheStorageFiles defaultCacheStorageFiles;
+
+  DefaultCacheStorageWriter(DefaultCacheStorageFiles defaultCacheStorageFiles) {
+    this.defaultCacheStorageFiles = defaultCacheStorageFiles;
+  }
+
+  /**
+   * Writes the {@link CacheWrite}.
+   *
+   * @param cacheWrite the {@link CacheWrite} to write out
+   * @return the {@link CacheEntry} representing the written entry
+   * @throws IOException if an I/O exception occurs
+   */
+  CacheEntry write(CacheWrite cacheWrite) throws IOException {
+    Path temporaryLayerFile = Files.createTempFile(null, null);
```

Comment:
I would expect this write to happen in the cache directory? I feel like mv is not free in Windows (but renames are?)